### PR TITLE
Mmr flush improvement

### LINKED ISSUE #1795: Improve mmr cache flush
### Prerequisites

* [ ] Are you running the latest version(s)?
* [ ] Did you check the [debugging guide](https://doc.subquery.network/academy/tutorials_examples/debug-projects.html#how-to-debug-a-subquery-project)?
* [ ] Did you check the [FAQs](https://doc.subquery.network/faqs/faqs.html) and [Discussions](https://github.com/subquery/subql/discussions)?
* [ ] Are you reporting to the correct repository?
* [ ] Did you search for an existing issue or pull request?

### Description

1. Mmr cache flush should be independent from fetch service storeCache flush
2. mmr currently flush every 6 seconds or 1000 records (if handler triggered and post process block), we want it flush by its own frequency and its own threshold. 6 seconds seems too many mmr records were created, and blocking db transaction.
3. Establish its own db connection, unblock from fetching service

### Details

These details can help to reproduce the environment the issue is occurring

**Local Environment:** [You can get this information from executing `subql version`.]
**Query Version:** [What is the version of the query service?]
**Indexer Version:** [What is the version of the indexer service?]
**Network Details:**
  * [Network]
  * [Block height, a block height where the issue is happening]
  * [Dictionary endpoint, if used]

### Steps to Reproduce

1. [First Step]
6. [Second Step]
7. [and so on...]

**Example project:** [A link to a minimal example that can reproduce the issue]

**Expected behavior:** [What you expected to happen]

**Actual behavior:** [What actually happened]

### Any other information

Is there any other information you would like to add?

### PR DESCRIPTION
**Pull request by jiqiang90**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #1795 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have tested locally
- [x] I have performed a self review of my changes
- [ ] Updated any relevant documentation
- [x] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] My code is up to date with the base branch
- [x] I have updated relevant changelogs. [We suggest using chan](https://github.com/geut/chan/tree/main/packages/chan)

### CODE REVIEW

File: packages/node-core/src/indexer/unfinalizedBlocks.service.spec.ts [hunk 1]
```diff
@@ -4,13 +4,12 @@
 // import { Header } from '@polkadot/types/interfaces';
 import {EventEmitter2} from '@nestjs/event-emitter';
 import {SchedulerRegistry} from '@nestjs/schedule';
-import {CacheMetadataModel} from './storeCache/cacheMetadata';
-import {StoreCacheService} from './storeCache/storeCache.service';
+import {StoreCacheService, CacheMetadataModel} from './storeCache';
```
[the remark this answers is not in the document]
Comment (jiqiang90) [reply packages/node-core/src/indexer/unfinalizedBlocks.service.spec.ts:7]: used here https://github.com/subquery/subql/blob/cd2ce872ebbd1932f58ca397d94a082033f41fcc/packages/node-core/src/indexer/unfinalizedBlocks.service.spec.ts#L62

File: packages/node-core/src/indexer/unfinalizedBlocks.service.spec.ts [hunk 2]
```diff
@@ -4,13 +4,12 @@
...
 import {
-  Header,
   METADATA_LAST_FINALIZED_PROCESSED_KEY,
   METADATA_UNFINALIZED_BLOCKS_KEY,
   BaseUnfinalizedBlocksService,
+  Header,
```
[the remark this answers is not in the document]
Comment (jiqiang90) [reply packages/node-core/src/indexer/unfinalizedBlocks.service.spec.ts:12]: https://github.com/subquery/subql/blob/cd2ce872ebbd1932f58ca397d94a082033f41fcc/packages/node-core/src/indexer/unfinalizedBlocks.service.spec.ts#L19

File: packages/node-core/src/indexer/storeCache/pgMmrCache.service.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    const schema = await getExistingProjectSchema(nodeConfig, this._sequelize);
+    assert(schema, 'Unable to check for MMR table, schema is undefined');
+    const db = await CachePgMmrDb.create(this.sequelize, schema);
+    this.setMmrRepo(this.sequelize, db);
+    return db;
+  }
+
+  setMmrRepo(sequelize: Sequelize, mmrRepo: CachePgMmrDb): void {
+    this._sequelize = sequelize;
+    this._mmrRepo = mmrRepo;
```
[the remark this answers is not in the document]
Comment (jiqiang90) [reply packages/node-core/src/indexer/storeCache/pgMmrCache.service.ts:58]: I think I would add a guard here first, and still finding solution for https://github.com/subquery/subql/issues/1842

File: packages/node-core/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+### Changed
+- Added `pgMmrCacheService` to use independent Db connection, now mmr cache flush by itself. (#1828)
 ### Changed
```
Comment (stwiname) [inline packages/node-core/CHANGELOG.md:11]:
```suggestion
### Changed
- Added `pgMmrCacheService` to use independent Db connection, now mmr cache flush by itself. (#1828)
```

**Review by stwiname — APPROVED** [review]

**Pull request merged**